frame/vesting/src/tests.rs

```diff
@@ -307,53 +513,644 @@ fn force_vested_transfer_works() {
+			1, // Vest over 2 * 256 blocks.
+			10,
+		);
+
+		// Add max amount schedules to user 4.
+		for _ in 0 .. max_schedules {
+			assert_ok!(Vesting::force_vested_transfer(RawOrigin::Root.into(), 13, 4, sched));
+		}
+
+		// The schedules count towards vesting balance.
+		let transferred_amount = <Test as Config>::MinVestedTransfer::get() * max_schedules as u64;
+		assert_eq!(Vesting::vesting_balance(&4), Some(transferred_amount));
+		// and free balance.
+		user_4_free_balance += transferred_amount;
+		assert_eq!(Balances::free_balance(&4), user_4_free_balance);
+
+		// Cannot insert a 4th vesting schedule when `MaxVestingSchedules` === 3
+		assert_noop!(
+			Vesting::force_vested_transfer(RawOrigin::Root.into(), 3, 4, sched),
+			Error::<Test>::AtMaxVestingSchedules,
+		);
+		// so the free balance does not change.
+		assert_eq!(Balances::free_balance(&4), user_4_free_balance);
+
+		// Account 4 has fully vested when all the schedules end,
+		System::set_block_number(<Test as Config>::MinVestedTransfer::get() + 10);
+		assert_eq!(Vesting::vesting_balance(&4), Some(0));
+		// and after unlocking its schedules are removed from storage.
+		vest_and_assert_no_vesting::<Test>(4);
+	});
+}
+
+#[test]
+fn merge_schedules_that_have_not_started() {
+	ExtBuilder::default().existential_deposit(ED).build().execute_with(|| {
+		// Account 2 should already have a vesting schedule.
+		let sched0 = VestingInfo::new(
+			ED * 20,
+			ED, // Vest over 20 blocks.
+			10,
+		);
+		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched0]);
+		assert_eq!(Balances::usable_balance(&2), 0);
+
+		// Add a schedule that is identical to the one that already exists
+		assert_ok!(Vesting::vested_transfer(Some(3).into(), 2, sched0));
+		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched0, sched0]);
+		assert_eq!(Balances::usable_balance(&2), 0);
+		assert_ok!(Vesting::merge_schedules(Some(2).into(), 0, 1));
+
+		// Since we merged identical schedules, the new schedule finishes at the same
+		// time as the original, just with double the amount
+		let sched1 = VestingInfo::new(
+			sched0.locked() * 2,
+			sched0.per_block() * 2,
+			10, // starts at the block the schedules are merged
+		);
+		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched1]);
+
+		assert_eq!(Balances::usable_balance(&2), 0);
+	});
+}
+
+#[test]
+fn merge_ongoing_schedules() {
+	// Merging two schedules that have started will vest both before merging.
+	ExtBuilder::default().existential_deposit(ED).build().execute_with(|| {
+		// Account 2 should already have a vesting schedule.
+		let sched0 = VestingInfo::new(
+			ED * 20,
+			ED, // Vest over 20 blocks
+			10,
+		);
+		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched0]);
+
+		let sched1 = VestingInfo::new(
+			ED * 10,
+			ED,                          // Vest over 10 blocks
+			sched0.starting_block() + 5, // Start at block 15
+		);
+		assert_ok!(Vesting::vested_transfer(Some(4).into(), 2, sched1));
```

Comment:
idea: you could probably clean up tests a bit by directly setting whatever vested transfer you need on any user, rather than repeating the process of transferring from some (arbitrary) account into your desired accounts. 

Might need to be its own PR.

for example I can foresee

```
ExtBuilder::default()
	.add_vest(2, 20 * ED, ED, 10)
	.add_vest(2, 10 * ED, ED, 15)
	.build()
```

Reply:
I agree with you, but I am not sure if I should do in this PR. I don't want to increase the diff in the tests I have not touched

Reply:
https://github.com/paritytech/substrate/issues/9345